pandas/core/window/rolling.py

```diff
@@ -542,12 +544,16 @@ def homogeneous_func(values: np.ndarray):
                 return values.copy()
 
             def calc(x):
+
                 start, end = window_indexer.get_window_bounds(
                     num_values=len(x),
                     min_periods=min_periods,
                     center=self.center,
                     closed=self.closed,
                 )
+
+                # From get_window_bounds, those two should be equal in length of array
+                assert len(start) == len(end)
```

Comment:
Could this assertion be moved to `GroupbyIndexer.get_window_bounds` instead?

Reply:
I'm fine with putting one there too -- I was just mirroring existing practice -- but I think an independent check here has merit.  

If start/end are broken here, `result = calc(values)` is probably going to crash a few lines below. 

Since there's logic in _apply which calculates min_periods which is passed as an arg to get_window_bounds, we can dream up scenarios where everything would be fine at that level but broken here.

Reply:
Okay. Putting this assert here is fine as well then